modules/platforms/dotnet/Apache.Ignite.Core/Services/ServiceCallContextBuilder.cs

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Apache.Ignite.Core.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text;
+    using Apache.Ignite.Core.Impl.Common;
+    using Apache.Ignite.Core.Impl.Services;
+
+    /// <summary>
+    /// Service call context builder.
+    /// </summary>
+    public class ServiceCallContextBuilder
+    {
+        /** Context attributes. */
+        private readonly Dictionary<string, byte[]> _attrs = new Dictionary<string, byte[]>();
+        
+        /// <summary>
+        /// Put string attribute.
+        /// </summary>
+        /// <param name="name">Attribute name.</param>
+        /// <param name="value">Attribute value.</param>
+        /// <returns>This for chaining.</returns>
+        public ServiceCallContextBuilder Put(string name, string value)
```

Comment:
To be consistent with `BinaryObjectBuilder`:

```suggestion
        public ServiceCallContextBuilder Set(string name, string value)
```

Reply:
Done, the same change has been made to java.

Reply:
Not sure about Java, where `Put` is idiomatic for Maps.
There is no problem if some APIs are named differently in Java and C#, I think it is more important to use language-specific idioms so that the users are more comfortable.

Reply:
ok, kept for java "put", thanks.